Gems/RecastNavigation/Code/CMakeLists.txt

```diff
@@ -19,6 +19,43 @@ o3de_pal_dir(pal_dir ${CMAKE_CURRENT_LIST_DIR}/Platform/${PAL_PLATFORM_NAME} "${
 # is supported by this platform.
 include(${pal_dir}/PAL_${PAL_PLATFORM_NAME_LOWERCASE}.cmake)
 
+# Add RecastNavigation dependency using FetchContent
+# Note: RecastNavigation::Detour requires lowering of the warning level to compile.
+include(FetchContent)
+FetchContent_Declare(
+        RecastNavigation
+        GIT_REPOSITORY "https://github.com/recastnavigation/recastnavigation"
+        GIT_TAG "5a870d427e47abd4a8e4ce58a95582ec049434d5" # main branch
+)
```

Comment:
This is pulling recast navigation, a third party library. Since this is being merged into development we have to be very careful of what 3rdParty libraries we are adding.

AFAIK this is not an official way to add 3rdParty library dependencies into O3DE, It's not following the normal procedures. For example, there was a lot of effort in putting all 3rdParty dependencies together to make sure each package specifies the license it's got and there is an script that collects all 3rdparty packages O3DE uses with their licenses. This way of adding 3rdParty dependency will be missed and it's not clarified anywhere what license it's under.

While it's nice that the gem pulls and compiles the code, we still have to make sure we follow the legal procedures of adding a 3rdParty library into O3DE.

What license "recast navigation" uses? If it's `Apache-2.0` only then you should be ok, but if it's got any other license then it needs to be reviewed and approved by TSC.

Tagging @lawsonamzn and @jeremyong

Reply:
Recast Navigation is not using Apache-2.0 license. https://github.com/recastnavigation/recastnavigation/blob/master/License.txt

Reply:
Thanks @AMZN-Olex for confirming the appropriate steps have been followed and approved. Great work!